### ISSUE TITLE
Plugin 'Lombok' (version '0.34.1-2019.1') is not compatible with Android Studio build AI-211.7628.21

### ISSUE THREAD

**Issue by ykrapiva** [mplushnikov/lombok-intellij-plugin#1107]
### Short description
After installing an update of Android Studio "Bumblebee" the plugin stopped working. It is not visible in the Plugin -> Marketplace, nor it is possible to install it from .zip.

### Expected behavior
There should be an update to the plugin

### Version information

* Android Studio Bumblebee | 2021.1.1
* Build #AI-211.7628.21.2111.8092744, built on January 19, 2022
* Runtime version: 11.0.11+9-b60-7590822 amd64
* VM: OpenJDK 64-Bit Server VM by Oracle Corporation
* Windows 10 10.0
* GC: G1 Young Generation, G1 Old Generation
* Memory: 3072M
* Cores: 12
* Registry: external.system.auto.import.disabled=true, debugger.watches.in.variables=false
* Non-Bundled Plugins: com.jetbrains.ChooseRuntime (1.2), org.jetbrains.kotlin (211-1.6.10-release-923-AS7442.40), org.intellij.plugins.markdown (211.7142.37)

### Additional information
Error message is: Plugin 'Lombok' (version '0.34.1-2019.1') is not compatible with the current version of the IDE, because it requires build 191.* or older but the current build is AI-211.7628.21

**Comment by ykrapiva** [comment]
Temporary fixed the 'until' version of the plugin manually, and then installed the plugin from a .zip file.
I followed the instructions from this comment: https://github.com/mplushnikov/lombok-intellij-plugin/issues/1028#issuecomment-937485680

**Comment by vvy6791** [comment]
I also upgraded today to Android Studio v. AI-211.7628.21
And my last lombok plugin also does not work now. So, I simply unistalled it. But IDEA does not "see" lombok features...

A quick look around shows, that [lombok plugin was add to Comunity version of IDEA from 2020.3.x. The IDE comes bundled with the plugin.](https://www.baeldung.com/lombok-ide)
AS Bumblebee based on IDEA 2021.1.1. It seems to be Google "fogot" to include this plugin to this version of Android Studio.

Workaroud:
1. [Download last Comunity distribution.](https://www.jetbrains.com/idea/download)
2. Extract from distribution folder **plugins/lombok** and put it to your ".../Android Studio/plugins/" folder.
3. Restart AS.
You will see that lombok plugin becomes "bundled".

**Comment by eriknyk** [comment]
any update, about when it will be fixed?

**Comment by shoyu666** [comment]
intellij idea 2021.2.3 build#ic-212.5457.46  
lombok plugin copy to AS works.
thanks @vvy6791

**Comment by giaur500** [comment]
So I guess this one is dead, last update year ago. Next Android Studio update will revert ”bundled" version

**Comment by vvy6791** [comment]
> workaround not works for me. I got:
> Plugin 'Lombok' (version '0.34.1-2019.1') is not compatible with the current version of the IDE, because it requires build 191.* or older but the current build is AI-211.7628.21

Did you unistall lombok plugin from AS? It must be done first.
And be careful when downloading Community Edition distribution with **your** IDEA version. To extract actual plugin version from it.

> Any update, about when it will be fixed?

I'm not from Google. And I don't know really. But I think they will fix it in the next update.

**Comment by giaur500** [comment]
> I'm not from Google. And I don't know really. But I think they will fix it in the next update.

But what will the fix exactly? They need to ship Lombok Plugin with Android Studio. Maybe they won't do that, because Lombok is usable for Java only and Java is not recommended language for Android - Kotlin is default.

**Comment by vvy6791** [comment]
> ... Maybe they [_Google_] won't do that, because Lombok is usable for Java only and Java is not recommended language for Android - Kotlin is default.
However, there are still quite a few Java developers for Android, who supprorts old Java projects and prefer Java in there job. I don't think Google wants to lose them.

**Comment by grgaaa** [comment]
> I also upgraded today to Android Studio v. AI-211.7628.21 And my last lombok plugin also does not work now. So, I simply unistalled it. But IDEA does not "see" lombok features...
> 
> A quick look around shows, that [lombok plugin was add to Comunity version of IDEA from 2020.3.x. The IDE comes bundled with the plugin.](https://www.baeldung.com/lombok-ide) AS Bumblebee based on IDEA 2021.1.1. It seems to be Google "fogot" to include this plugin to this version of Android Studio.
> 
> Workaroud:
> 
> 1. [Download IDEA distribution Comunity Edition v. **2021.1.1**.](https://www.jetbrains.com/idea/download/other.html)
> 2. Extract from distribution folder **plugins/lombok** and put it to your ".../Android Studio/plugins/" folder.
> 3. Restart AS.
>    You will see that lombok plugin becomes "bundled".

This worked for me. I just wanted to add that I needed to install the latest distribution of IDEA 2021.1, being 2021.1.3.
The lombok plugin included in the base version 2021.1 also triggered the error. @vvy6791 Thank you for the workaround. Let's hope google releases a patch soon.

**Comment by eriknyk** [comment]
@grgaaa it works only on comunity edition? 
I've tried my self, but I have a ultimate version because I have the license and it doesn't worked for me.

**Comment by grgaaa** [comment]
@eriknyk I don't have the ultimate version of IntelliJ so I can't confirm or deny if this is a problem. Personally I doubt that the ultimate version has a different bundled Lombok plugin but can't be really sure. Did you also try installing v. 2021.1.1? I guess the main rule to make it work is that the build number of IDEA and AndroidStudio match. 
In my case: 
- IntelliJ Idea: Version: 2021.1.3 Build: 211.7628.21
- AndroidStudio: Version 2021.1.1 Build: AI-**211.7628.21**.2111.8092744

**Comment by Mikhael-Danilov** [comment]
> But I think they will fix it in the next update.

Looks like they removed lombok intentionally. https://issuetracker.google.com/issues/202423443

**Comment by eriknyk** [comment]
But independently that if it was removed intentionally or not as a default bundle, is it not possible to fix the compatibility with the newest version of android studio to have to installed manually from the plugins manager?

**Comment by basilbattikhi** [comment]
Do you have any update on this issue ?

**Comment by giaur500** [comment]
> > But I think they will fix it in the next update.
> 
> Looks like they removed lombok intentionally. https://issuetracker.google.com/issues/202423443

Do you know the reason? Url lincluded there is therir internal system which I can't access. 

> But independently that if it was removed intentionally or not as a default bundle, is it not possible to fix the compatibility with the newest version of android studio to have to installed manually from the plugins manager?

I guess not, because standalone version is abandoned. I have no idea what's their problem with Lombok and why they decided to don't bundle it with Android Studio

**Comment by Mikhael-Danilov** [comment]
> Do you know the reason? Url lincluded there is their internal system which I can't access.

I have no idea. Only another clue I found so far: https://issuetracker.google.com/issues/184071996

**Comment by unenergizer** [comment]
> Temporary fixed the 'until' version of the plugin manually, and then installed the plugin from a .zip file. I followed the instructions from this comment: [#1028 (comment)](https://github.com/mplushnikov/lombok-intellij-plugin/issues/1028#issuecomment-937485680)

This worked for me. It was as simple as extracting the xml file, making the edit, and putting it back into the jar file. Thank you!

**Comment by ashish-d-hh** [comment]
> Temporary fixed the 'until' version of the plugin manually, and then installed the plugin from a .zip file. I followed the instructions from this comment: [#1028 (comment)](https://github.com/mplushnikov/lombok-intellij-plugin/issues/1028#issuecomment-937485680)

This worked for me, temporary fix

**Comment by eriknyk** [comment]
Is this plugin currently abandoned?
I can see that last version uploaded to jetbrains plugin repository was from jan 2021

**Comment by philiip11** [comment]
If the above solution doesn't work for you, here is an alternative way:
You can use the attached Zip-File and skip to step 5 or create it yourself starting with step 1.
[lombok-plugin.zip](https://github.com/mplushnikov/lombok-intellij-plugin/files/7993248/lombok-plugin.zip)

1. Download IntelliJ IDEA Community Edition [2021.1.3 - Windows ZIP Archive (zip)](https://download.jetbrains.com/idea/ideaIC-2021.1.3.win.zip)
2. copy the folder plugins/lombok to another location
3. rename that folder to lombok-plugin
4. create a Zip-File and rename that to lombok
5. Open Android Studio, press Ctrl + Alt + S (or open settings via file menu), go to 
plugins
6. Click on the settings gear icon right to installed
7. Select "Install Plugin from Disk..."
8. Select the zip file and press ok.
9. Lombok is installed
10. Profit

**Comment by shipofsea** [comment]
> I also upgraded today to Android Studio v. AI-211.7628.21 And my last lombok plugin also does not work now. So, I simply unistalled it. But IDEA does not "see" lombok features...
> 
> A quick look around shows, that [lombok plugin was add to Comunity version of IDEA from 2020.3.x. The IDE comes bundled with the plugin.](https://www.baeldung.com/lombok-ide) AS Bumblebee based on IDEA 2021.1.1. It seems to be Google "fogot" to include this plugin to this version of Android Studio.
> 
> Workaroud:
> 
> 1. [Download IDEA distribution Comunity Edition v. **2021.1.1**.](https://www.jetbrains.com/idea/download/other.html)
> 2. Extract from distribution folder **plugins/lombok** and put it to your ".../Android Studio/plugins/" folder.
> 3. Restart AS.
>    You will see that lombok plugin becomes "bundled".

thanks
it works for me

**Comment by shipofsea** [comment]
you can download Lombok 0.34.1-2019.1 zip file at the below url
https://plugins.jetbrains.com/plugin/6317-lombok/versions/stable/108259

**Comment by ykrapiva** [comment]
> you can download Lombok 0.34.1-2019.1 zip file at the below url https://plugins.jetbrains.com/plugin/6317-lombok/versions/stable/108259

Right, then edit it and install as mentioned in https://github.com/mplushnikov/lombok-intellij-plugin/issues/1028#issuecomment-937485680

**Comment by XilinJia** [comment]
Further from @DilipDalwadi , I downloaded lombok-plugin-0.34-2020.2.zip from marketplace, unzip it, went to the META-INF directory, changed in the xml file the line to <idea-version since-build="191.6183" until-build="IE-212.*"/>, then saved the zip, installed the plugin from disk on Android Studio.  Things seem to work - previous errors regarding unresolved getter functions now gone.  Pretty brute force.  Not sure if there will be side effects, but who knows if this plugin will only be updated after next year?

**Comment by johannbandion** [comment]
Any updates how long it will take to update

**Comment by winstone-su** [comment]
> 

It make useful!

**Comment by DevDengChao** [comment]
I've uploaded two version of this plugin to #1111, which works with latest Android Studio (`Bumblebee 2021.1.1 Patch 2` and `Dolphin 2021.3.1`), if anyone don't want to repeat the step of download IDEA, then you can try the files in #1111 .

**Comment by mciobanu** [comment]
> @grgaaa it works only on comunity edition? I've tried my self, but I have a ultimate version because I have the license and it doesn't worked for me, so the only way to get it working was rollback the android studio to the previos version artic fox.

I also have Ultimate (IU-213.6777.52). I had the plugin working fine before the update, and seen as incompatible after.

I uninstalled the plugin from AS, copied the plugin's folder from Ultimate to AS, restarted AS, and all seems fine. The plugin is viewed as bundled.

**Comment by Naccl** [comment]
https://stackoverflow.com/a/71025235/13403382

**Comment by TristanK27** [comment]
> https://stackoverflow.com/a/71025235/13403382
> 
> This worked for me.

Same here. By far the simplest solution for me.

**Comment by HaoyuWang00** [comment]
Hi there, if you are still looking for a solution, you can have a look at this [post](https://stackoverflow.com/a/73349686/13773355). Hope this helps!

**Comment by ElectroBuddha** [comment]
I can confirm that this is working for latest Android Studio Chipmunk | 2021.2.1 Patch 2 on MacBook Pro (Apple M1)
https://www.youtube.com/watch?v=A14hrtUeNWU

**Comment by robertoferreirajrr** [comment]
Thanks @ElectroBuddha life saver!

**Comment by FreedomLabsIO** [comment]
> I can confirm that this is working for latest Android Studio Chipmunk | 2021.2.1 Patch 2 on MacBook Pro (Apple M1) https://www.youtube.com/watch?v=A14hrtUeNWU

Works perfectly with Android Studio AI-231.* for Mac M1. Thanks
